crates/miden-lib/asm/kernels/transaction/lib/account.masm

```diff
@@ -468,6 +468,27 @@ export.get_item
     # => [VALUE]
 end
 
+#! Gets an item from the account storage at its initial state (beginning of transaction).
+#!
+#! Note:
+#! - We assume that index has been validated and is within bounds.
+#!
+#! Inputs:  [index]
+#! Outputs: [INIT_VALUE]
+#!
+#! Where:
+#! - index is the index of the item to get.
+#! - INIT_VALUE is the initial value of the item at the beginning of the transaction.
+export.get_item_init
+    # get account initial storage slots section offset
+    exec.memory::get_account_initial_storage_slots_ptr
+    # => [account_initial_storage_slots_ptr, index]
+
+    # get the item from initial storage
+    swap mul.8 add padw movup.4 mem_loadw
+    # => [INIT_VALUE]
```

Comment:
I'd maybe move this logic into a `get_item_raw` procedure and then call it from `get_item_init` and `get_item`. 

The interface should roughly look like:

```
#! Inputs:  [storage_slots_ptr, index]
#! Outputs: [INIT_VALUE]
export.get_item_init_raw
    # ...
end
```

And then we can remove `account_delta::get_item_initial` and use `get_item_init_raw` wherever we used it instead.